devdoc/channel_internal.md

```diff
@@ -30,15 +30,15 @@ typedef void(*PUSH_CALLBACK)(void* push_context, CHANNEL_CALLBACK_RESULT result)
 ```
 THANDLE_TYPE_DECLARE(CHANNEL_INTERNAL);
 
-MOCKABLE_FUNCTION(, THANDLE(CHANNEL_INTERNAL), channel_internal_create_and_open, THANDLE(THREADPOOL), threadpool);
+MOCKABLE_FUNCTION(, THANDLE(CHANNEL_INTERNAL), channel_internal_create_and_open, THANDLE(PTR(LOG_CONTEXT_HANDLE)), log_context, THANDLE(THREADPOOL), threadpool);
 MOCKABLE_FUNCTION(, void, channel_internal_close, THANDLE(CHANNEL_INTERNAL), channel_internal);
```

Comment:
>channel_internal_close [](http://example.com/codeflow?start=26&length=22)

why is this API needed?

Looks like "once closed it stays closed" in which case, it serves no further purpose so it might as well release its used resources (aka "destroy").

Suspicion that it needs to go away.

Reply:
See "Need for `CHANNEL_INTERNAL`"  in channel.md.

`CHANNEL_INTERNAL` needs to be closed when the user releases the last reference to `THANDLE(CHANNEL)`